androidNode/src/androidMain/kotlin/network/bisq/mobile/android/node/service/offers/NodeOffersServiceFacade.kt

```diff
@@ -291,19 +327,63 @@ class NodeOffersServiceFacade(
     }
 
     private fun isValidOfferbookMessage(message: BisqEasyOfferbookMessage): Boolean {
-        // TODO Add more validation
-        // In Bisq main we have that code in the bisqEasyOfferbookMessageService.isValid(message)
-        // method
-        /*
-         public boolean isValid(BisqEasyOfferbookMessage message) {
+        // Mirrors Bisq main: see bisqEasyOfferbookMessageService.isValid(message)
         return isNotBanned(message) &&
-                isNotIgnored(message) &&
-                (isTextMessage(message) ||
-                        isBuyOffer(message) ||
-                        hasSellerSufficientReputation(message));
+            isNotIgnored(message) &&
+            (isTextMessage(message) || isBuyOffer(message) || hasSellerSufficientReputation(message))
+    }
+
+    private fun isNotBanned(message: BisqEasyOfferbookMessage): Boolean {
+        val authorUserProfileId = message.authorUserProfileId
+        val profileOpt = userProfileService.findUserProfile(authorUserProfileId)
+        if (!profileOpt.isPresent) {
+            log.w { "isNotBanned: author profile not found for id=$authorUserProfileId" }
+            return false
+        }
+        return !bannedUserService.isUserProfileBanned(profileOpt.get())
```

Comment:
We have that method in the service: `isUserProfileBanned(String userProfileId)`, thus the findUserProfile and handling of not found profiles is not needed to replicate here (with the risk of impl a diff. policy).